### PR TITLE
[webnn] Add float32 tests for WebNN convTranspose2d op.

### PR DESCRIPTION
**Pull request by BruceDai**
The PR of [`WebNN convTranspose2d`](https://webmachinelearning.github.io/webnn/#api-mlgraphbuilder-convtranspose2d) op tests based on both [convTranspose2d tests](https://github.com/webmachinelearning/webnn-baseline/blob/7507aa385faeb0eb57120ab9a6aa2f9f2a559b54/test/conv_transpose2d_test.js) and [`convTranspose2d` implementation](https://github.com/webmachinelearning/webnn-baseline/blob/7507aa385faeb0eb57120ab9a6aa2f9f2a559b54/src/conv_transpose2d.js) (for generating **float64** `input data (including weights, bias)` and **float32** `expected data`) by https://github.com/webmachinelearning/webnn-baseline/pull/31.

@fdwr @domenic @Honry PTAL, thanks.

### CODE REVIEW

**Review by fdwr — APPROVED** [review]

File: webnn/resources/utils.js [hunk 1]
```diff
@@ -94,14 +95,20 @@ const getConv2dPrecisionTolerance = (resources, operationName) => {
       inputChannels = options.inputLayout === 'nchw' ? inputChannels : inputShape[3];
     }
     if (options.filterLayout) {
-      if (!['oihw', 'hwio', 'ohwi', 'ihwo'].includes(options.filterLayout)) {
+      let filterLayouts = ['oihw', 'hwio', 'ohwi', 'ihwo']; // default for conv2d
```
Comment (Honry) [inline webnn/resources/utils.js:98]: Nit: s/let/const

Comment (BruceDai) [reply webnn/resources/utils.js:98]: @Honry This variable `filterLayouts` will be set to a new value from next Line-99 as following:
```js
      if (operationName === 'convTranspose2d') {
        filterLayouts = ['iohw', 'hwoi', 'ohwi'];
      }
```
PTAL, thanks.

**Comment by BruceDai** [comment]
The new commit is to update tests to align with #38276, @Honry PTAL, thanks.

File: webnn/resources/utils.js [hunk 2]
```diff
@@ -81,7 +81,8 @@ const getConv2dPrecisionTolerance = (resources, operationName) => {
   const options = resources.options;
   let groups = 1;
   let inputChannels = inputShape[1]; // default nchw inputLayout
-  let filterWidth = filterShape[3]; // default oihw filterLayout
+   // default oihw filterLayout for conv2d or default iohw filterLayout for convTranspose2d
```
Comment (Honry) [inline webnn/resources/utils.js:84]: Nit: inconsistent indention.

File: webnn/resources/test_data/conv_transpose2d.json [hunk 3]
```diff
@@ -0,0 +1,1951 @@
...
+            -59.3402779768406,
+            50.17738650272881,
+            -72.58010817012352,
+            48.844270540309566
+          ],
+          "type": "float32"
+        }
+      },
+      "options": {
+        "activation": "relu" // relu operator
```
[the remark this answers is not in the document]
Comment (fdwr) [reply webnn/resources/test_data/conv_transpose2d.json:1975]: He strips them before parsing in utils.js loadTests:

`const resources = JSON.parse(json.replace(/\\"|"(?:\\"|[^"])*"|(\/\/.*|\/\*[\s\S]*?\*\/)/g, (m, g) => g ? "" : m));`

[the remark this answers is not in the document]
Comment (Honry) [reply webnn/resources/test_data/conv_transpose2d.json:1975]: Oh, thanks, acked.

File: webnn/resources/utils.js (hunk 2 above)
Comment (BruceDai) [reply webnn/resources/utils.js]: Done, thanks.

File: webnn/conv_transpose2d.https.any.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+// META: global=window,dedicatedworker
+// META: script=./resources/utils.js
+// META: script=./webnn-polyfill.js
+// META: timeout=long
+
+'use strict';
+
+// https://webmachinelearning.github.io/webnn/#api-mlgraphbuilder-convtranspose2d
+
+const buildConvTranspose2d= (operationName, builder, resources) => {
```
[the remark this answers is not in the document]
Comment (BruceDai) [reply webnn/conv_transpose2d.https.any.js]: Done, thanks.

File: webnn/conv_transpose2d.https.any.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+// META: title=test WebNN API convTranspose2d operation
+// META: global=window,dedicatedworker
+// META: script=./resources/utils.js
+// META: script=./webnn-polyfill.js
```
[the remark this answers is not in the document]
Comment (BruceDai) [reply webnn/conv_transpose2d.https.any.js]: Already done. Thanks.

**Review by Honry — APPROVED** [review]

**Pull request merged**